nodeadm/internal/feature-gates/feature-gates.go

```diff
@@ -0,0 +1,18 @@
+package featuregates
+
+type FeatureGate = string
+
+const (
+	// InstanceIdNamedNode When this flag is enabled, override the hostname with instanceId
+	InstanceIdNamedNode = FeatureGate("instanceIdNamedNode")
```

Comment:
im curious, should we further abstract these to avoid thinking about the mode in several places?

something like 
```golang
type FeatureGate struct {
  Key string
  Verifier func(map[string]bool) bool
}

func (fg FeatureGate) IsEnabled(featureGates map[FeatureGate]bool) bool {
  return fg.Verifier(fg.Key, featureGates)
}

var InstanceIdNamedNode = FeatureGate{
  Key: "instanceIdNamedNode",
  Verifier: DefaultTrue,
}
```

Reply:
good idea, updated

Reply:
this works for now, but its also just a suggestion. not sure if we'll need more complicated handling like differences across versions